tests/Unit/EmogrifierTest.php

```diff
@@ -690,6 +690,118 @@ public function emogrifyNotAppliesCssToNonMatchingElements($css, $expectedHtml)
         static::assertContains(sprintf($expectedHtml, $cssDeclaration1, $cssDeclaration2), $result);
     }
 
+    /**
+     * @return string[][]
+     */
+    public function differentCssSelectorSpecificityDataProvider()
+    {
+        return [
+            'class more specific than types' => ['html body p', '.p-1', '<p class="p-1"'],
+            'ID more specific than class and types' => ['html body .p-4', '#p4', '<p class="p-4" id="p4"'],
+            'ID more specific than 20 classes and types' => [
+                'html body ' . str_repeat('.p-4', 20),
+                '#p4',
+                '<p class="p-4" id="p4"'
+            ],
+            ':not with type more specific than nothing' => ['.p-1', '.p-1:not(h1)', '<p class="p-1"'],
+            ':not with several types more specific than single type' => ['p', '*:not(html body h1)', '<p class="p-1"'],
+            ':not with class more specific than types' => ['html body p', '*:not(.p-2)', '<p class="p-1"'],
+            ':not with class more specific than :not with types' => [
+                '*:not(html body h1)',
+                '*:not(.p-2)',
+                '<p class="p-1"'
+            ],
+            'class more specific than :not with types' => ['*:not(html body h1)', '.p-1', '<p class="p-1"'],
+        ];
+    }
+
+    /**
+     * @test
+     * @param string $lessSpecificSelector A selector expression
+     * @param string $moreSpecificSelector Some other, more specific selector expression
+     * @param string $matchedTagPart Tag expected to be matched by both selectors, without the closing '>',
+     *                               e.g. '<p class="p-1"'
+     * @dataProvider differentCssSelectorSpecificityDataProvider
+     */
+    public function emogrifyAppliesMoreSpecificCssSelectorToMatchingElements(
+        $lessSpecificSelector,
+        $moreSpecificSelector,
+        $matchedTagPart
+    ) {
+        $this->subject->setHtml(static::COMMON_TEST_HTML);
+        $this->subject->setCss(
+            $lessSpecificSelector . ' { color: red } ' .
+            $moreSpecificSelector . ' { color: green } ' .
+            $moreSpecificSelector . ' { background-color: green } ' .
+            $lessSpecificSelector . ' { background-color: red }'
```

Comment:
Any particular reason why you're leaving out the closing semicolon (`color: red;`)?

Reply:
No particular reason.  Have added it.